ktor-network/jvm/test/io/ktor/network/sockets/tests/UdpBroadcastTest.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.sockets.tests
+
+import io.ktor.network.selector.*
+import io.ktor.network.sockets.*
+import io.ktor.util.network.*
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.debug.junit4.*
+import org.junit.*
+import org.junit.Test
+import org.junit.rules.*
+import java.util.concurrent.*
+import kotlin.coroutines.*
+import kotlin.test.*
+
+class UdpBroadcastTest : CoroutineScope {
+    private val testJob = Job()
+    private val exec = Executors.newCachedThreadPool()
+    private val selector = ActorSelectorManager(exec.asCoroutineDispatcher() + testJob)
+
+    override val coroutineContext: CoroutineContext
+        get() = testJob
+
+    @get:Rule
+    val timeout = CoroutinesTimeout.seconds(15)
+
+    @get:Rule
+    val errors = ErrorCollector()
+
+    @After
+    fun tearDown() {
+        testJob.cancel()
+
+        selector.close()
+        exec.shutdown()
+    }
+
+    @Test
+    fun testBroadcast() {
+        val serverSocket = CompletableDeferred<BoundDatagramSocket>()
+        val server = launch(Dispatchers.Default) {
+            try {
+                aSocket(selector)
+                    .udp()
+                    .bind(NetworkAddress("0.0.0.0", 56700))
+                    .use { socket ->
+                        serverSocket.complete(socket)
+                        val received = socket.receive()
+                        assertEquals("0123456789", received.packet.readText())
+                    }
+            } catch (t: Throwable) {
```

Comment:
cause

Reply:
Please see the other PR: https://github.com/ktorio/ktor/pull/2075